File: content-hunter/scripts/format_bilibili.py
# -*- coding: utf-8 -*-
import json
import os
from datetime import datetime

# 读取原始数据
with open('E:/workspace/content-hunter/data/bilibili_raw.json', 'r', encoding='utf-8') as f:
    items = json.load(f)

print(f'共有 {len(items)} 条B站数据')

# 格式化播放量
def format_num(s):
    s = str(s)
    if '万' in s:
        return s
    try:
        n = int(s)
        if n >= 10000:
            return f'{n/10000:.1f}万'
        return str(n)
    except:
        return s

# 格式化标题（去除em标签）
def clean_title(title):
    import re
    title = re.sub(r'<em class="keyword">', '', title)
    title = re.sub(r'</em>', '', title)
    return title

# 追加写入md
md_path = 'E:/workspace/content-hunter/data/bilibili.md'
today = datetime.now().strftime('%Y-%m-%d %H:%M')

file_exists = os.path.exists(md_path)
with open(md_path, 'a', encoding='utf-8') as f:
    if not file_exists:
        f.write('# B站AI技术热门内容\n\n')
    
    f.write(f'## 抓取时间: {today} (本次共 {len(items[:100])} 条)\n\n')
    
    for i, item in enumerate(items[:100], 1):
        title = clean_title(item.get('title', ''))
        author = item.get('author', '')
        play = item.get('play', '0')
        video_review = item.get('video_review', '0')
        favorites = item.get('favorites', '0')
        like = item.get('like', '0')
        duration = item.get('duration', '00:00')
        bvid = item.get('bvid', '')
        description = item.get('description', '')
        arcurl = item.get('arcurl', f'https://www.bilibili.com/video/{bvid}')
        
        f.write(f'### 第{i}条\n')
        f.write(f'- 标题: {title}\n')
        f.write(f'- UP主: {author}\n')
        f.write(f'- 播放: {format_num(play)}\n')
        f.write(f'- 弹幕: {format_num(video_review)}\n')
        f.write(f'- 点赞: {format_num(like)}\n')
        f.write(f'- 收藏: {format_num(favorites)}\n')
        f.write(f'- 时长: {duration}\n')
        f.write(f'- BV号: {bvid}\n')
        f.write(f'- 链接: {arcurl}\n')
        desc_text = description[:200] if description else '暂无简介'
        f.write(f'- 内容总结: {desc_text}\n\n')

print(f'已追加写入 {min(len(items), 100)} 条到 {md_path}')

File: content-hunter/data/format_bilibili.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('E:/workspace/content-hunter/data/bilibili_raw.json', 'utf8'));

function stripHtml(text) {
    return String(text || '').replace(/<[^>]+>/g, '').trim();
}

const items = data.map(v => {
    const score = (v.play || 0) * 0.1 + (v.like || 0) * 2 + (v.favorites || 0) * 3 + (v.review || 0);
    return {
        title: stripHtml(v.title),
        author: v.author || '未知',
        play: v.play || 0,
        danmaku: v.video_review || 0,
        like: v.like || 0,
        coins: 0,
        favorites: v.favorites || 0,
        review: v.review || 0,
        duration: v.duration || '未知',
        tags: (v.tag || '').split(',').filter(t => t).slice(0, 5).join(' #'),
        url: v.arcurl || '',
        bvid: v.bvid || '',
        desc: stripHtml(v.description).substring(0, 200),
        score: score
    };
}).sort((a, b) => b.score - a.score);

const top100 = items.slice(0, 100);

let header = '# B站热门AI技术内容\n\n';
header += '> 抓取时间: ' + new Date().toLocaleString('zh-CN', { timeZone: 'Asia/Shanghai' }) + ' | 数据来源: Bilibili搜索 | 关键词: AI技术/人工智能/大模型/ChatGPT/深度学习 | 本次: ' + data.length + '条 → TOP100\n\n';

let md = '';
for (let i = 0; i < top100.length; i++) {
    var item = top100[i];
    md += '### 第' + (i+1) + '条\n';
    md += '- 标题: ' + item.title + '\n';
    md += '- UP主: ' + item.author + '\n';
    md += '- 播放: ' + item.play.toLocaleString() + '\n';
    md += '- 弹幕: ' + item.danmaku.toLocaleString() + '\n';
    md += '- 点赞: ' + item.like.toLocaleString() + '\n';
    md += '- 收藏: ' + item.favorites.toLocaleString() + '\n';
    md += '- 时长: ' + item.duration + '\n';
    md += '- 话题: #' + item.tags + '\n';
    md += '- 链接: ' + item.url + '\n';
    md += '- 内容总结: ' + item.desc + '\n\n';
}

const outPath = 'E:/workspace/content-hunter/data/bilibili.md';
const isFirstWrite = !fs.existsSync(outPath);
const existing = isFirstWrite ? '' : fs.readFileSync(outPath, 'utf8');
fs.writeFileSync(outPath, existing + '\n' + (isFirstWrite ? header : '') + md);
console.log('[' + (isFirstWrite ? 'NEW FILE' : 'APPEND') + '] Wrote ' + top100.length + ' items to ' + outPath);
console.log('Top 3 by score:');
top100.slice(0, 3).forEach((v, i) => console.log((i+1) + '. ' + v.title.substring(0, 50) + ' | 播放:' + v.play.toLocaleString() + ' 点赞:' + v.like.toLocaleString()));
